AppCenterData/AppCenterData/Internal/Client/MSDataOperationProxy.m

```diff
@@ -239,17 +245,21 @@ - (void)updateLocalStore:(MSTokenResult *)token
     [self.documentStore deleteWithToken:token documentId:currentCachedDocument.documentId];
   }
 
-  /*
-   * If the cached document has a create or replace pending operation, and no eTags, and if the current operation is a
-   * deletion, delete the document from the store.
-   */
+   //If the cached document has a create or replace pending operation, and no eTags, delete the document from the store.
```

Comment:
So if I do two offline updates, would that delete the document from local storage? We need the current operation to be a deletion in order to delete the local document from storage.

Reply:
Right, we need to keep the last condition and add the new one like you did below.